lib/fileSystem2.ts

```diff
@@ -1,17 +1,29 @@
 import * as fs from "fs";
 import { getParentFolderPath, getPathName, joinPath } from "./path";
 
+async function _entryExists(entryPath: string, condition?: (stats: fs.Stats) => (boolean | Promise<boolean>)): Promise<boolean> {
+  return new Promise((resolve, reject) => {
+    fs.lstat(entryPath, (error: NodeJS.ErrnoException, stats: fs.Stats) => {
```

Comment:
This promisification could be useful in this library https://github.com/ts-common/fs/tree/master/src